src/source/callback.rs

```diff
@@ -0,0 +1,55 @@
+use std::marker::PhantomData;
+use std::time::Duration;
+
+use crate::{Sample, Source};
+
+/// A source which execute a callback function
+pub struct Callback<S> {
+    pub phantom_data: PhantomData<S>,
+    pub callback: Box<dyn Send + Fn()>,
+}
+
+impl<S> Callback<S> {
+    #[inline]
+    pub fn new(callback: Box<dyn Send + Fn()>) -> Callback<S> {
+        Callback {
+            phantom_data: PhantomData,
+            callback,
+        }
+    }
+}
+
+impl<S> Iterator for Callback<S> {
+    type Item = S;
+
+    #[inline]
+    fn next(&mut self) -> Option<S> {
+        (self.callback)();
+        None
+    }
+}
+
+impl<S> Source for Callback<S>
+where
+    S: Sample,
+{
+    #[inline]
+    fn current_frame_len(&self) -> Option<usize> {
+        None
+    }
+
+    #[inline]
+    fn channels(&self) -> u16 {
+        1
+    }
+
+    #[inline]
+    fn sample_rate(&self) -> u32 {
+        48000
```

Comment:
Why is this not adjustable?

Reply:
cargo culted from https://github.com/RustAudio/rodio/blob/master/src/source/empty.rs

i imagine it doesn't actually matter since this source never contains audio

Reply:
Fair, I rest my case.